File: miniJobs/UI/minijobs_mobile/lib/models/job/job_save_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:minijobs_mobile/models/job/job_schedule_info.dart';

part 'job_save_request.g.dart';

@JsonSerializable()
class JobSaveRequest {
  int? id;
  String? name;
  String? description;
  String? streetAddressAndNumber;
  int? cityId;
  int? status;
  int? jobTypeId;
  int? requiredEmployees;
  JobScheduleInfo? jobSchedule;
  Map<int, List<int>>? answersToPaymentQuestions;
  int? wage;
  int? applicationsDuration;

  JobSaveRequest(
      this.id,
      this.name,
      this.description,
      this.streetAddressAndNumber,
      this.cityId,
      this.status,
      this.jobTypeId,
      this.requiredEmployees,
      this.jobSchedule,
      this.answersToPaymentQuestions,
      this.wage,
      this.applicationsDuration);

  factory JobSaveRequest.fromJson(Map<String, dynamic> json) =>
      _$JobSaveRequestFromJson(json);

  Map<String, dynamic> toJson() => _$JobSaveRequestToJson(this);
}

File: miniJobs/Application/Jobs/Models/JobSaveRequest.cs
using Domain.Enums;

namespace Application.Jobs.Models;

public class JobSaveRequest
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string StreetAddressAndNumber { get; set; }
    public int CityId { get; set; }
    public JobStatus Status { get; set; }
    public int? JobTypeId{ get; set; }
    public JobScheduleInfo? JobSchedule { get; set; }
    public Dictionary<int, List<int>?> AnswersToPaymentQuestions { get; set; }
    public int? RequiredEmployees { get; set; }
    public string? Description { get; set; }
    public int? Wage { get; set; }
    public int? ApplicationsDuration { get; set; }
}